jsonschema-core.xml

```diff
@@ -800,49 +800,257 @@
+                        keywords not present in any vocabulary, beyond what the meta-schemas
+                        associated with the declared vocabularies describe.  This allows for
+                        restricting usage to a subset of a vocabulary, and for validating
+                        locally defined keywords not intended for re-use.
+                    </t>
+                    <t>
+                        Meta-schemas intended for local use, with no need to test for
+                        vocabulary support in arbitrary implementations, can safely omit
+                        "$vocabulary" entirely.
+                    </t>
+                </section>
+                <section title="The JSON Schema Core Vocabulary">
+                    <t>
+                        Keywords declared in in this specification that begin with "$" make up
+                        the JSON Schema Core vocabulary.  These keywords are either required in
+                        order process any schema or meta-schema, including those split across
+                        multiple documents, or exist to reserve keywords for purposes that
+                        require guaranteed interoperability.
+                    </t>
+                    <t>
+                        The Core vocabulary MUST be considered mandatory at all times, in order
+                        to bootstrap the processing of further vocabularies.  Meta-schema
+                        authors MAY explicitly list the Core vocabulary, but do not need
+                        to do so.  If present, the Core vocabulary entry MUST have a value
+                        of true indicating that it is required.  The behavior of a value
+                        of false is undefined.
+                    </t>
+                    <t>
+                        The current URI for the Core vocabulary is:
+                        <eref target="https://json-schema.org/draft-08/vocabularies/core"/>.
+                    </t>
+                    <t>
+                        The current URI for the corresponding meta-schema is:
+                        <eref target="https://json-schema.org/draft-08/core"/>.
+                    </t>
+                </section>
+                <section title="Example Meta-Schema With Vocabulary Declarations">
+                    <figure>
+                        <preamble>
+                            This meta-schema explicitly declares both the Core and Applicator
+                            vocabularies, and combines their meta-schemas with an "allOf".
+                            It additional restricts the usage of the Applicator vocabulary
+                            by forbidding the keyword prefixed with "unevaluated".  It also
+                            describes a keyword, "localKeyord", that is not part of either
+                            vocabulary.  Note that it is its own meta-schema,
```

Comment:
> Note that it is its own meta-schema...

This is a nice thing to have for this example.  Can we use the same language to describe the same idea for *applicators.json* and *core.json*?  Not sure where it'd go, but it seems necessary.

Reply:
I don't actually understand what you're asking here.  The example `$ref`s both the core and applicator meta-schemas, so it is already validating `$vocabulary` which is present in the core vocabulary's meta-schema.

I'm not sure what language you are looking to re-use from here to those other meta-schemas.  They are neither restricting vocabulary terms nor are they describing extra keywords.  Nor are either of them their own meta-schemas (they require keywords like `type`, for example).  So these four lines do not seem to apply.

Reply:
Wait... is it its own meta-schema because it declares itself in `$schema`?  If that's the case, then it has to link back to the *draft-08* meta-schemas somehow, which is the purpose of `$vocabulary`.  It's like saying, "This meta-schema is self-validating, but also must validate against these other meta-schemas."

I think I'm starting to get this.

Reply:
@gregsdennis yes!

There's some convoluted bootstrapping, and a certain amount of "yeah, probably everyone needs to understand some minimal vocabulary set" that we'll work out.

Meta-schemas are not vocabularies.  And vocabularies are not meta-schemas.  So you will never see a vocabulary URI in `$schema`, nor will you see a meta-schema URI in `$vocabulary`.

They collect vocabularies to inform an implementation what they're doing, because it's really quite complex to infer such a thing out of the meta-schema itself.  It's not exactly impossible but it's hard.

In the case of the example, it happens to work that it only needs core and applicator, and the extra constraints (restricting `unevaluated*` and describing a locally defined keyword) don't get in the way of what it needs.  So its `$id` and `$schema` are the same, and neither are directly related to the contents of `$vocabulary`.

Reply:
@gregsdennis I attempted to address this in the most recent version, so I'm marking this thread as resolved and we can start a new conversation attached to the current spec lines for whatever I did not manage to address fully.